java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java

```diff
@@ -1028,4 +1036,81 @@ public void testRemoveSpecificAPIpatterns() throws IOException {
             "}")
         .doTest();
   }
+
+  public void transformAndCreateNewPropertyFile(
+      String srcProp, String trgtProp, String stale_flag, boolean isTreated) throws IOException {
+
+    String[] temp = stale_flag.split("_");
+    String flagNameCamelCase =
+        temp[0]
+            + Arrays.stream(temp, 1, temp.length)
+                .map(str -> str.substring(0, 1).toUpperCase() + str.substring(1))
+                .collect(joining());
+    String newContent =
+        Files.readAllLines(Paths.get(srcProp))
+            .stream()
+            .map(x -> x.replace("$flagName$", flagNameCamelCase))
+            .map(x -> x.replace("$notIsTreated$", String.valueOf(!isTreated)))
+            .collect(joining("\n"));
+    Path trgt = Paths.get(trgtProp);
+    Files.deleteIfExists(trgt);
+    Files.write(trgt, newContent.getBytes(UTF_8), StandardOpenOption.CREATE);
+  }
+
+  @Test
+  public void testMethodChainTreated() throws IOException {
```

Comment:
Can we add one more test with the same properties.json without referring `allow_method_chain` but having `.` in the methodName references in the json. Then, ensure the tests will result in no refactoring. This can be in a followup too.

Reply:
This is something which is a bit difficult to address because we have an independent visitor in XPFlagCleaner that deletes the abstract method from the interface based on the annotation match, irrespective of whether the allow chain option is set to true or false. The main concern here is that, the resulting code will contain a compilation error (abstract method is deleted, while invocations are not) which is hard to produce in the way we test through EP's APIs.

Reply:
Ok. Maybe it is not completely unchanged and the definitions get deleted. The expected output can just have those deletions without the chain based deletions. There is an option for `allowBreakingChanges` with the refactoring that should handle any compilation errors that gets introduced due to this.